backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/TimeMachineService.java

```diff
@@ -117,88 +116,84 @@ public Norm applyPassiveModifications(ApplyPassiveModificationsUseCase.Query que
   }
 
   record ModData(Optional<Href> targetHref, Optional<Href> targetUpToRef, Mod mod) {}
 
-  private void applyQuotedText(final ModData modData, Node targetNode) {
-    final Mod mod = modData.mod();
-    if (mod.getSecondQuotedText().isEmpty()
-        || mod.getOldText().isEmpty()
-        || mod.getNewText().isEmpty()) return;
-    String oldText = mod.getOldText().get();
-    String newText = mod.getNewText().get();
-
-    if (mod.containsRef()) {
-      final Node secondQuotedTextNode = mod.getSecondQuotedText().get();
-
-      // Get the target node's current text content and remove unnecessary whitespaces, for the
-      // character counting from mod to work
-      final String targetText = targetNode.getTextContent().trim().replaceAll("\\s+", " ");
-      if (targetText.isEmpty()) return;
-
-      // Get the start and end index of the old text, within the cleaned old text of the target node
-      final int startIndex = targetText.indexOf(oldText);
-      final int endIndex = startIndex + oldText.length();
-
-      // Split the text into three parts: before, replacement, after
-      final String beforeText = targetText.substring(0, startIndex);
-      final String afterText = targetText.substring(endIndex + 1);
-
-      // Clear the target node's content
-      while (targetNode.hasChildNodes()) {
-        targetNode.removeChild(targetNode.getFirstChild());
-      }
+  private void applyMod(final ModData modData, final Norm targetZf0Norm) {
+    if (modData.targetHref().isEmpty() || modData.targetHref().get().getEId().isEmpty()) {
+      return;
+    }
 
-      // Import beforeText
-      if (StringUtils.isNotEmpty(beforeText)) {
-        final Text beforeTextNode = targetNode.getOwnerDocument().createTextNode(beforeText);
-        targetNode.appendChild(beforeTextNode);
-      }
+    final var targetEid = modData.targetHref().get().getEId().get();
+    final var targetNode =
+        NodeParser.getNodeFromExpression(
+            String.format("//*[@eId='%s']", targetEid), targetZf0Norm.getDocument());
+
+    if (targetNode.isEmpty()) {
+      return;
+    }
 
-      // Import content of quotedText from amending law (which include akn:refs) using importNode
-      // because of different documents
-      NodeParser.nodeListToList(secondQuotedTextNode.getChildNodes())
-          .forEach(
-              child -> {
-                final Node importedChild = targetNode.getOwnerDocument().importNode(child, true);
-                targetNode.appendChild(importedChild);
-              });
-
-      // Import afterText
-      if (StringUtils.isNotEmpty(afterText)) {
-        final Text afterTextNode = targetNode.getOwnerDocument().createTextNode(afterText);
-        targetNode.appendChild(afterTextNode);
+    if (modData.mod().getSecondQuotedText().isPresent()) {
+      try {
+        applyQuotedText(modData, targetNode.get());
+      } catch (IllegalArgumentException | IndexOutOfBoundsException exception) {
+        log.info(
```

Comment:
Shouldn't this be on error level? Or is it info level because the time machine runs always even if the mod is not complete/correct?

Reply:
As this is something we expect to happen regularly (for the reasons you mentioned) i was thinking that info is fine